components/google_sheets/sources/new-row-added/new-row-added.js

```diff
@@ -121,19 +136,16 @@ module.exports = {
     const { headers } = event;
     if (!headers) return;
 
-    let sheetIDs = this.sheetIDs || [];
     if (headers["x-goog-resource-state"] === "sync") {
       // initialize row counts
-      if (sheetIDs.length === 0) {
-        const sheets = (
-          await this.google_drive.listSheets(
-            this.drive === "myDrive" ? null : this.drive
-          )
-        ).options;
-        for (const s of sheets) sheetIDs.push(s.value);
-      }
-      const rowCounts = await this.google_sheets.getRowCounts(sheetIDs);
+      const rowCounts = await this.google_sheets.getRowCounts(this.sheetID);
```

Comment:
Maybe rename the method to `getWorksheetRowCounts`? I was initially confused at why there would be multiple counts for a given spreadsheet.